### ISSUE TITLE
How to submit more transaction in concurrency and wait for them to be committed?

### ISSUE THREAD

**Issue by marfvr** [cosmos/ethermint-archive#385]
`ethermint version 0.5.3`
I need that `eth_sendTransaction` requests as the following can be submitted in concurrency and wait until each of them is committed:

        curl -X POST --data '{"params": [ 
            {"from": "0x7eff122b94897ea5b0e2a9abf47b86337fafebdc", 
            "to": "0xb1c1e8c408491862e35fadc47613e965b810a046",
            "value": "0x1"}], 
            "method": "eth_sendTransaction", "id": 3, "jsonrpc": "2.0"}' 
            http://172.57.101.100:8545

Where `http://172.57.101.100:8545` is the RPC listen address of an Ethermint instance.

In other words, I need a mechanism to submit transactions and to wait until they are committed.

Ethermint implementation uses the `broadcast_tx_sync` API of Tendermint, which is blocking only until the `CheckTx` returns. I tried to replace the call from `broadcast_tx_sync` to `broadcast_tx_commit` but now only one transaction at a time is picked and validated into a block, and in the meanwhile no other request is considered.

Any suggestion?

**Comment by zramsay** [comment]
till will be addressed in the new ethermint: https://github.com/cosmos/ethermint
